### DISCUSSION TITLE
How to properly reference button name in string

### DISCUSSION THREAD

**Discussion by sabuncu** [WeblateOrg/weblate#5290] [Q&A]
Disclaimer - I am new to weblate.

I came across the following string: 

> Please enter the installation key for {0} V{1}, then click "Next".

My question: Obviously, "Next" is the name of a button.  But why is it hardcoded in the string? I would think you would need to use a reference to the button, which would then resolve to the button's translated name in the target language.  Am I missing something?

Thanks.

**Comment by nijel** [comment]
This really depends on how the software is localized. It would be certainly possible to use format string for the button name, but the person writing the code would have to realize that it's useful. In many cases this isn't obvious and asking the developers to use that will do the trick. Sometimes it's not.even possible due to technical limitations (for example when translating documentation) though...

**Reply by sabuncu** [reply]
Thank you for your response.

If this approach were to be used, would you maintain a list of button names?  For example, in a WinForms project, the button name would be a resource in the project.  When a new locale is implemented, that resource name would be displayed in its translated form.  Then, you can also use that resource in a format string referring to that button.

> Sometimes it's not.even possible due to technical limitations (for example when translating documentation) though...

I am trying to understand the sentence above.  Is it not possible because documentation does not have access to the resource names used in code?  If so, would it work if documentation was part of the overall software package, and could reference the resource names used in the code?

Thank you again.
